newlib/libc/sys/vita/io.c

```diff
@@ -132,6 +132,35 @@ int __vita_release_descriptor(int fd)
 	return 0;
 }
 
+int __vita_duplicate_descriptor(int fd)
+{
+	int fd2 = -1;
+
+	sceKernelLockLwMutex(&_newlib_fd_mutex, 1, 0);
+
+	if (!is_fd_valid(fd))
+	{
+		sceKernelUnlockLwMutex(&_newlib_fd_mutex, 1);
```

Comment:
I've added commit cd0ed97 to this PR cleaning up the mutex paths as suggested. I favoured inverting the logic checks over goto.